dags/sentinel-2_data_transfer.py

```diff
@@ -83,36 +85,68 @@ def get_self_link(message_content):
     for link in message_content["links"]:
         if link["rel"] == "canonical":
             self_link = link["href"]
+    return self_link
 
-    return self_link.replace(
-        "https://sentinel-cogs.s3.us-west-2.amazonaws.com",
-        f"s3://{SRC_BUCKET_NAME}",
+
+def get_src_link(message_content):
+    for link in message_content["links"]:
+        if link["rel"] == "derived_from":
+            return link["href"]
+
+
+def correct_stac_link(metadata):
+    """
+    Replace the https link of the source bucket with s3 link of the destination bucket
+    """
+
+    # Extrac source link before updating STAC file
+    src_link = get_self_link(metadata)
+    # Drop canonical and via-cirrus links
+    metadata["links"] = [
+        x
+        for x in metadata["links"]
+        if x["rel"] != "canonical" and x["rel"] != "via-cirrus"
+    ]
+
+    # Update links to match destination
+    stac = json.dumps(metadata)
+    "Replace https with s3 uri"
+    stac = json.loads(
+        stac.replace(
+            "https://sentinel-cogs.s3.us-west-2.amazonaws.com",
+            f"s3://deafrica-sentinel-2",
+        )
     )
+    # Update source link
+    links = [x for x in stac["links"] if x["rel"] == "derived_from"]
+    for x in stac["links"]:
+        # Replace derived-from link with s3 links to sentinel-cogs bucket
+        if x["rel"] == "derived_from":
+            x["href"] = src_link.replace(
+                "https://sentinel-cogs.s3.us-west-2.amazonaws.com",
+                f"s3://sentinel-cogs",
+            )
 
+    return stac
 
-def publish_to_sns_topic(message):
+
+def publish_to_sns_topic(updated_stac, message_attributes):
```

Comment:
Naming things is important. This function publishes a message with attributes to sns.
```suggestion
def publish_to_sns(message, attributes):
```